src/solver/expressions/include/antares/solver/expressions/visitors/SubstitutionVisitor.h

```diff
@@ -0,0 +1,48 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+
+#include <antares/solver/expressions/Registry.hxx>
+#include <antares/solver/expressions/nodes/NodesForwardDeclaration.h>
+#include <antares/solver/expressions/visitors/CloneVisitor.h>
+
+namespace Antares::Solver::Visitors
+{
+struct SubstitutionContext
+{
+    std::vector<Nodes::ComponentVariableNode*> variables;
```

Comment:
if the ComponentVariableNode::id is unique, variables could be an unordered_map, faster and with direct access

Reply:
I now use an `std::unordered_set`

Reply:
Actually it's a bit more complicated. `std::set::find` only works if we lookup elements, there is no `std::set::find_if`. So we have to use `std::find_if` that has linear complexity. I left a comment in the code, so we can come back to it later if need be.